### DISCUSSION TITLE
Missing data for flopy tutorials

### DISCUSSION THREAD

**Discussion by JJHelly** [modflowpy/flopy#2391] [Q&A] [answered]
Aloha.  I'm trying to run the flopy tutorials and cannot get past this missing data problem.  Seems like something basic is wrong in that the code expects ../../examples/data to be installed with the appropriate input files but they are not there.  I have tried to find them everywhere I can think to look but have no success.  Any pointers would be most welcome.  Mahalo.  

Here's an example of the problem attached in the pdf of the jupyter output.

**Comment by wpbonelli** [comment]
@JJHelly some tutorials and examples rely on data which can be found in the repository under the path you mention. Unfortunately this assumes the whole repository has been checked out and that you are running the notebooks from the folder they are in. Ideally the notebooks would all be self contained, so they could run immediately after download from the documentation site. We have that on the roadmap https://github.com/modflowpy/flopy/issues/1872.

**Comment by JJHelly** [comment]
Mahalo for explanation. I can try that
Sent from my iPhone

On Dec 7, 2024, at 14:51, wpbonelli ***@***.***> wrote:

﻿

@JJHelly<https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!HUMjnxaS7j1dOXYQUtHvh2kxsSE_gX1DQeJXXPeB_yOUHz1oMisV9wJXgMEZqsuP6IGWWlYrZlI07C8W4qPaWo3o$> some tutorials and examples rely on data which can be found in the repository under the path you mention. Unfortunately this assumes the whole repository has been checked out and that you are running the notebooks from the folder they are in. Ideally the notebooks would all be self contained, so they could run immediately after download from the documentation site. We have that on the roadmap #1872<https://urldefense.com/v3/__https://github.com/modflowpy/flopy/issues/1872__;!!Mih3wA!HUMjnxaS7j1dOXYQUtHvh2kxsSE_gX1DQeJXXPeB_yOUHz1oMisV9wJXgMEZqsuP6IGWWlYrZlI07C8W4hoSAPsm$>.

**Comment by JJHelly** [comment, marked as answer]
Mahalo.

I was able to fix the problem by copying the examples directory, however, I don't understand how anybody can use the tutorials.  Am I the only one who encountered this problem?

Also, even when I cloned the who repo, there was no place that I detected where you could run the noteboooks from.  Granted, I am a relative newb wrt python and its conventions but seems like this is a big hurdle for most.  What am I missing, please?

J.
----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Dec 7, 2024, at 14:51, wpbonelli ***@***.***> wrote:



@JJHelly<https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!HUMjnxaS7j1dOXYQUtHvh2kxsSE_gX1DQeJXXPeB_yOUHz1oMisV9wJXgMEZqsuP6IGWWlYrZlI07C8W4qPaWo3o$> some tutorials and examples rely on data which can be found in the repository under the path you mention. Unfortunately this assumes the whole repository has been checked out and that you are running the notebooks from the folder they are in. Ideally the notebooks would all be self contained, so they could run immediately after download from the documentation site. We have that on the roadmap #1872<https://urldefense.com/v3/__https://github.com/modflowpy/flopy/issues/1872__;!!Mih3wA!HUMjnxaS7j1dOXYQUtHvh2kxsSE_gX1DQeJXXPeB_yOUHz1oMisV9wJXgMEZqsuP6IGWWlYrZlI07C8W4hoSAPsm$>.

**Reply by wpbonelli** [reply]
@JJHelly I am sure you are not the only one. This is a long-standing complaint. Perhaps the tutorials were written with the assumption people would simply read and copy/paste from them. But ideally they should run out of the box (given a suitable Python environment and Modflow executables) with no expectation regarding the working directory or contents of the filesystem. Hope to have this sorted out soon.

**Comment by JJHelly** [comment]
Aloha.

Well, I can offer something that might be helpful.  I think all you really have to do is to provide a place on your tutorial workflow for downloading the examples directory.

That's what I was looking for and there isn't anything except some mis-leading reference on other web-sites that, for example, the Freyberg stuff is part of the 'distribution', presumably the code distribution; which it is not.

You don't have to go to the trouble of bundling all the examples with their individual requirements if they all can be pointed at the examples directory; wherever the user puts it on their machine.

Mahalo for you good work, BTW.

J.
----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Dec 8, 2024, at 04:14, wpbonelli ***@***.***> wrote:



@JJHelly<https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!DOpdmarKFyMI2GA4PZCziNSSgFT7GvFbwxpzE331Rs8fudLy6fFHDQcscNETp8ur4PDf8BtQX5kOaONL2XEQy6J1$> I am sure you are not the only one. This is a long-standing complaint. Perhaps the tutorials were written with the assumption people would simply read and copy/paste from them. But ideally they should run out of the box (given a suitable Python environment and Modflow executables) with no expectation regarding the working directory or contents of the filesystem. Hope to have this sorted out soon.

**Comment by wpbonelli** [comment]
With https://github.com/modflowpy/flopy/commit/7298cd1cca36134bb2d72f8a8c7d5489aa004488 the notebooks should now work wherever they are run from, and whether or not data preexists locally

**Comment by JJHelly** [comment]
Mahalo.

This is a nice improvement.

J.
----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Dec 12, 2024, at 04:51, wpbonelli ***@***.***> wrote:



With 7298cd1<https://urldefense.com/v3/__https://github.com/modflowpy/flopy/commit/7298cd1cca36134bb2d72f8a8c7d5489aa004488__;!!Mih3wA!DDCieRrpX4-SbPMhDQlOdJathajRvJNkVXLLD8v8iLSzsIdrzZ548PS7oteKuputYKoa-n3DNUCxz2pbQPGAGTWe$> the notebooks should now work wherever they are run from, and whether or not data preexists locally

**Comment by JJHelly** [comment]
Aloha.

I'm trying to run the Synthetic Valley example but run into what appears to be an error in the model (rather than the computing environment).   I'm not sure where to report this type of error, so would you please advise?   I don't want to bother you with this stuff.

Console output follows:


 '                U.S. GEOLOGICAL SURVEY MODULAR HYDROLOGIC MODEL',
 '                            VERSION 6.5.0 05/23/2024',
 '',
 '        MODFLOW 6 compiled May 23 2024 18:00:56 with GCC version 13.2.0',
 '',
 'This software has been approved for release by the U.S. Geological ',
 'Survey (USGS). Although the software has been subjected to rigorous ',
 'review, the USGS reserves the right to update the software as needed ',
 'pursuant to further analysis and review. No warranty, expressed or ',
 'implied, is made by the USGS or the U.S. Government as to the ',
 'functionality of the software and related material nor shall the ',
 'fact of release constitute any such warranty. Furthermore, the ',
 'software is released on condition that neither the USGS nor the U.S. ',
 'Government shall be held liable for any damages resulting from its ',
 'authorized or unauthorized use. Also refer to the USGS Water ',
 'Resources Software User Rights Notice for complete use, copyright, ',
 'and distribution information.',
 '',
 '',
 ' MODFLOW runs in SEQUENTIAL mode',
 '',
 ' Run start date and time (yyyy/mm/dd hh:mm:ss): 2024/12/12 12:14:56',
 '',
 ' Writing simulation list file: mfsim.lst',
 ' Using Simulation name file: mfsim.nam',
 '',
 '',
 'ERROR REPORT:',
 '',
 '  1. CELL (3,6300) THICKNESS <= 0. TOP, BOT:    -15.2401388700000',
 '     -15.2401388700000',
 '',
 'UNIT ERROR REPORT:',
 '',
 "  1. ERROR OCCURRED WHILE READING FILE 'flow.disv'"]
----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Dec 12, 2024, at 04:51, wpbonelli ***@***.***> wrote:



With 7298cd1<https://urldefense.com/v3/__https://github.com/modflowpy/flopy/commit/7298cd1cca36134bb2d72f8a8c7d5489aa004488__;!!Mih3wA!DDCieRrpX4-SbPMhDQlOdJathajRvJNkVXLLD8v8iLSzsIdrzZ548PS7oteKuputYKoa-n3DNUCxz2pbQPGAGTWe$> the notebooks should now work wherever they are run from, and whether or not data preexists locally

**Reply by christianlangevin** [reply]
We think this problem is new and due to a change in how numpy behaves on Windows and Mac.  We are testing a fix for it now.  See if you can get your version to work by changing

```
bot_l2[idomain_2] = -50.0 * ft2m
```

to 

```
bot_l2[idomain_2 == -1] = -50.0 * ft2m
```

Apologies for this.  We should have caught this earlier, but it snuck through the cracks.

**Comment by JJHelly** [comment]
Mahalo for the quick attention.  I will make the change and report back.  It occurred to me after the fact that this could be a precision problem since top and bottom are equal to many decimal places.
J.
------------------------------------------------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Dec 13, 2024, at 04:41, langevin-usgs ***@***.***> wrote:



We think this problem is new and due to a change in how numpy behaves on Windows and Mac. We are testing a fix for it now. See if you can get your version to work by changing

bot_l2[idomain_2] = -50.0 * ft2m


to

bot_l2[idomain_2 == -1] = -50.0 * ft2m


Apologies for this. We should have caught this earlier, but it snuck through the cracks.
